src/lib/starRelease.js

```diff
@@ -65,21 +65,36 @@ function generateLinearWithdrawTxs(contracts, amount, to) {
 }
 
 export async function getConditional(contracts, address) {
-  const [commitment, batches] = await Promise.all([
+  const [commitment, batches, remaining] = await Promise.all([
     conditionalSR.getCommitment(contracts, address),
     conditionalSR.getBatches(contracts, address),
+    conditionalSR.getRemainingStars(contracts, address),
   ]);
 
   if (commitment === null || batches === null) {
     return { total: 0, available: 0, withdrawn: 0, batchLimits: [] };
   }
 
-  const { total } = commitment;
-  const withdrawLimits = await Promise.all(
-    batches.map((_, idx) =>
-      conditionalSR.getWithdrawLimit(contracts, address, idx)
-    )
-  );
+  const forfeited = await conditionalSR.getForfeited(contracts, address);
+  const total = remaining.length;
+  let balance = total;
+
+  const withdrawLimits = await batches.reduce(async (p, _, idx) => {
```

Comment:
I don't think doing the balances for the batches in sequence like this (and potentially saying "not enough balance for this one batch") is entirely accurate, since you can withdraw from any batch: you don't necessarily empty them in order. Probably not a huge deal to pretend that's the case though.

Reply:
the UI doesn't support out of order withdrawal so it's kinda moot